src/hub/interfaces/IHubRequestManager.sol

```diff
@@ -0,0 +1,345 @@
+// SPDX-License-Identifier: BUSL-1.1
```

Comment:
I think this file should only contain the strictly common stuff used in all hub request managers by the core. Which are only `request()`, `claimDeposit()` and `claimRedeem()`. Other methods are no longer mandatory by the core, and the rest of the interface will be in `src/vaults`, along with the implementation

Reply:
Agree. I wonder if we could even remove `claimDeposit/Redeem` from `Hub`, and just expose these directly in `HubRequestManager`.

Reply:
I also thought about that when reviewing this, I agree!

Reply:
> Agree. I wonder if we could even remove `claimDeposit/Redeem` from `Hub`, and just expose these directly in `HubRequestManager`.

IIUC, we would still have to relay the request callbacks then via the hub in order to not introduce the hub request manager as ward to the message dispatcher, right?  Also in favor of this change to reduce `Hub` and improve gas cost of notifications by reducing contract calls. 

Curious if you intend to call `notify` directly or still via `hub.callRequestManager`. In case of the former, requires adding `ReentrancyProtection` to `HubRequestManager` to add `protected` for the notify calls.

Reply:
> IIUC, we would still have to relay the request callbacks then via the hub in order to not introduce the hub request manager as ward to the message dispatcher, right? Also in favor of this change to reduce Hub and improve gas cost of notifications by reducing contract calls.

Yeah like this is already the case for approveDeposits/Redeems etc.

> Curious if you intend to call notify directly or still via hub.callRequestManager. In case of the former, requires adding ReentrancyProtection to HubRequestManager to add protected for the notify calls.

I think adding protected modifier here would be good, yes.